P/PAPI/build_tarballs.jl

```diff
@@ -101,7 +104,7 @@ products = [
 #     components/cuda/cupti_common.c:114:23: error: ‘CUPTIU_MAX_FILES’ undeclared (first use in this function)
 #          char *found_files[CUPTIU_MAX_FILES];
 #                            ^
-cuda_platforms = CUDA.supported_platforms(; max_version=v"12.3")
+cuda_platforms = CUDA.supported_platforms(; max_version=v"12.3.999")
```

Comment:
Yeah, I noticed the mistake after the fact, but I feel like the `max_version` doesn't work as it should, I don't think it should exclude that version.

Reply:
Change the argument to `version_filter`, and pass `version_filter=version -> version < v"12.4"`, that's unambiguous. This will also allow for `version < v"12.4" || version >= v"12.5"` in the future.